components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java

```diff
@@ -803,4 +806,112 @@ public void testUpdateUserWithUsernameChange() throws Exception {
+                mockClaimMetadataManagementService, tenantDomain);
+        mockStatic(ApplicationManagementService.class);
+        when(ApplicationManagementService.getInstance()).thenReturn(applicationManagementService);
+        when(applicationManagementService.getServiceProvider(anyString(), anyString())).thenReturn(null);
+        scimUserManager.deleteUser(userId);
+        //this method is for testing of throwing CharonException, hence no assertion
+    }
+
+    @Test(expectedExceptions = CharonException.class)
+    public void testDeleteUserWithUserStoreDomainMismatch() throws Exception {
+
+        String userId = "12345";
+        String tenantDomain = "carbon.super";
+        Map<String, String> scimToLocalClaimsMap = new HashMap<>();
+        scimToLocalClaimsMap.put(SCIMConstants.CommonSchemaConstants.ID_URI, "userIdURI");
+        org.wso2.carbon.user.core.common.User coreUser = new org.wso2.carbon.user.core.common.User();
+        coreUser.setUserID(userId);
+        coreUser.setUsername("coreUser");
+        coreUser.setUserStoreDomain("PRIMARY");
+        List<org.wso2.carbon.user.core.common.User> coreUsers = new ArrayList<>();
+        coreUsers.add(coreUser);
+
+        mockStatic(SCIMCommonUtils.class);
+        when(SCIMCommonUtils.getSCIMtoLocalMappings()).thenReturn(scimToLocalClaimsMap);
+        AbstractUserStoreManager mockedUserStoreManager = PowerMockito.mock(AbstractUserStoreManager.class);
+        when(mockedUserStoreManager.getUserListWithID(anyString(), anyString(), anyString())).thenReturn(coreUsers);
+        SCIMUserManager scimUserManager = new SCIMUserManager(mockedUserStoreManager,
+                mockClaimMetadataManagementService, tenantDomain);
+        InboundProvisioningConfig inboundProvisioningConfig = new InboundProvisioningConfig();
+        inboundProvisioningConfig.setProvisioningUserStore("SECONDARY");
+        ServiceProvider serviceProvider = new ServiceProvider();
+        serviceProvider.setInboundProvisioningConfig(inboundProvisioningConfig);
+        mockStatic(ApplicationManagementService.class);
+        when(ApplicationManagementService.getInstance()).thenReturn(applicationManagementService);
+        when(applicationManagementService.getServiceProvider(anyString(), anyString())).thenReturn(serviceProvider);
+        scimUserManager.deleteUser(userId);
+        //this method is for testing of throwing CharonException, hence no assertion
+    }
+
+    @Test(expectedExceptions = AbstractCharonException.class)
+    public void testDeleteMe() throws Exception {
+
+        String userName = "testUser";
+        String tenantDomain = "carbon.super";
+        SCIMUserManager scimUserManager = new SCIMUserManager(mockedUserStoreManager,
+                mockClaimMetadataManagementService, tenantDomain);
+        when(mockedUserStoreManager.getUserIDFromUserName(userName)).thenThrow(new UserStoreException());
+        scimUserManager.deleteMe(userName);
```

Comment:
According to the method signature, this test seems to be the successful Me resource deletion. What is the reason for throwing a UserStoreException here?

Reply:
Addressed with commit [9050269](https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/350/commits/90502695675fdf01a56643a373e32695752f0ee4)
Wrote the test case to check the exception thrown while getting the user Id. So, updated the method signature.